src/transformers/models/fuyu/image_processing_fuyu.py

```diff
@@ -59,125 +219,332 @@ class FuyuImageProcessor(BaseImageProcessor):
+                Dictionary in the format `{"height": int, "width": int}` specifying the size of the patches.
         """
         requires_backends(self, ["torch"])
+        patch_size = patch_size if patch_size is not None else self.patch_size
+        patch_height, patch_width = patch_size["height"], patch_size["width"]
 
         # TODO refer to https://github.com/ArthurZucker/transformers/blob/0f0a3fe5ca5697ee58faeb5b53f049af720b5e98/src/transformers/models/vit_mae/modeling_vit_mae.py#L871
         # torch implementation is faster but does not handle non-squares
 
         batch_size, channels, _, _ = image.shape
-        unfolded_along_height = image.unfold(2, self.patch_size['height'], self.patch_size['height'])
-        patches = unfolded_along_height.unfold(3, self.patch_size["width"], self.patch_size["width"])
-
-        patches_reshaped = patches.contiguous().view(
-            batch_size, channels, -1, self.patch_size['height'], self.patch_size["width"])
-
-        patches_final = patches_reshaped.permute(0, 2, 3, 4, 1).reshape(
-            batch_size, -1, channels * self.patch_size['height'] * self.patch_size["width"]
-        )
-
-        return patches_final
-
-    def preprocess(self, images):
-        """Utility function to preprocess the images and extract necessary information about original formats."""
-        batch_images = []
-        image_unpadded_heights = []
-        image_unpadded_widths = []
-        images = make_list_of_images(images)
-        for image in images:
-            image = to_numpy_array(image)
-            if not is_scaled_image(image):
-                image = image / 255.0
-            channel_dimension = infer_channel_dimension_format(image, 3)
-            if channel_dimension == ChannelDimension.FIRST:
-                width_index = 2
-                height_index = 1
-            elif channel_dimension == ChannelDimension.LAST:
-                width_index = 1
-                height_index = 0
-
-            image_unpadded_widths.append([image.shape[width_index]])
-            image_unpadded_heights.append([image.shape[height_index]])
-
-            # Reproduct adept padding sampler
-            padded_image = self.scale_pad_normalize(image)
-            tensor_img = torch.Tensor(padded_image).permute(2, 0, 1)
-            batch_images.append([tensor_img])
-
-        return batch_images, torch.Tensor(image_unpadded_heights), torch.Tensor(image_unpadded_widths)
-
-    def postprocess_with_tokenizer_info(
+        unfolded_along_height = image.unfold(2, patch_height, patch_height)
+        patches = unfolded_along_height.unfold(3, patch_width, patch_width)
+        patches = patches.contiguous()
+        patches = patches.view(batch_size, channels, -1, patch_height, patch_width)
+        patches = patches.permute(0, 2, 3, 4, 1)
+        patches = patches.reshape(batch_size, -1, channels * patch_height * patch_width)
+        return patches
+
+    def preprocess_with_tokenizer_info(
```

Comment:
This was renamed to `preprocess_with_tokenizer_info` to reflect the current naming patterns with other image processors: `preprocess` for creating the model inputs, `post_process_xxx` for processing the model outputs for a specific downstream task

Reply:
Ok, good to know! thanks for the explanation